onadata/apps/logger/management/commands/update_attachment_storage_bytes.py

```diff
@@ -0,0 +1,147 @@
+                    # `attachment_storage_bytes` is being calculated.
+                    user_profile.metadata['submissions_suspended'] = True
+                    user_profile.save(update_fields=['metadata'])
+
+                # if `last_xform` is not none, it means that the `user_id` is
+                # different from the previous one in the loop so the user
+                # profile must be updated
+                if last_xform:
+                    self.update_user_profile(
+                        last_xform, total_per_user[last_xform['user_id']]
+                    )
+
+            # write out xform progress
+            if self.verbosity >= 1:
+                self.stdout.write(
+                    f"Calculating attachments for xform_id #{xform['pk']}"
+                    f" (user {xform['user__username']})"
+                )
+            # aggregate total media file size for all media per xform
+            form_attachments = Attachment.objects.filter(
+                instance__xform_id=xform['pk']
+            ).aggregate(total=Sum('media_file_size'))
+
+            if form_attachments['total']:
+                with transaction.atomic():
+                    if self.verbosity >= 1:
+                        self.stdout.write(
+                            f'\tUpdating xform attachment storage to '
+                            f"{form_attachments['total']} bytes"
+                        )
+
+                    XForm.objects.filter(
+                        pk=xform['pk']
+                    ).update(
+                        attachment_storage_bytes=form_attachments['total']
+                    )
+
+                total_per_user[xform['user_id']] += form_attachments['total']
+            elif self.verbosity >= 1:
+                self.stdout.write('\tNo attachments found')
+
+            last_xform = xform
+
+        # need to call `update_user_profile()` one more time outside the loop
+        # because the last user profile will not be up-to-date otherwise
+        if last_xform:
+            self.update_user_profile(
+                last_xform, total_per_user[last_xform['user_id']]
+            )
+
+        if self.verbosity >= 1:
+            self.stdout.write('Done!')
+
+    def update_user_profile(self, xform: dict, total: int):
+        user_id = xform['user_id']
+        username = xform['user__username']
+
+        if self.verbosity >= 1:
+            self.stdout.write(
+                f'Updating attachment storage total ({total} bytes) on '
+                f'{username}’s profile'
+            )
+
+        # Update user's profile (and lock the related row)
+        with transaction.atomic():
+            updates = {
+                'submissions_suspended': False,
+                'attachments_counting_status': 'complete',
+            }
+            UserProfile.objects.filter(
+                user_id=user_id
+            ).update(
+                attachment_storage_bytes=total,
```

Comment:
Wait, wouldn't this omit the totals from forms that were already complete before the management command started? It can be tricky to get this stuff right; I'd prefer to let Postgres handle the job by using the aggregate sum of all the user's forms as the new value for the user profile.

Reply:
Nope! The total is calculated from scratch, even for those already calculated in previous run.